torch/csrc/autograd/python_engine.cpp

```diff
@@ -61,6 +61,10 @@ variable_list PythonEngine::execute(
     bool keep_graph,
     bool create_graph,
     const edge_list& outputs) {
+  TORCH_CHECK(!PyGILState_Check(), "The autograd engine is called while holding the GIL. If you are using the C++ "
```

Comment:
nit: this reads a little better in past tense "The autograd engine was called" or "You called the autograd engine"